stdlib/contrib/anaisdg/statsmodels/linearreg.flux

```diff
@@ -1,10 +1,12 @@
+// Package statsmodels provides code for calculating linear regression.
 package statsmodels
 
-
 import "math"
 import "generate"
 
-// performs linear regression, calculates y_hat, and residuals squared (rse) 
+// linearRegression performs linear regression. It calculates and returns
+// [*ŷ*](https://en.wikipedia.org/wiki/Hat_operator#Estimated_value) (`y_hat`),
+// and [residual sum of errors](https://en.wikipedia.org/wiki/Residual_sum_of_squares) (`rse`).
```

Comment:
~~~suggestion
// linearRegression performs a linear regression.
//
// It calculates and returns [*ŷ*](https://en.wikipedia.org/wiki/Hat_operator#Estimated_value) (`y_hat`),
// and [residual sum of errors](https://en.wikipedia.org/wiki/Residual_sum_of_squares) (`rse`).
// Output data includes the following columns:
//
// - **N**: number of points in the calculation.
// - **slope**: slope of the calculated regression.
// - **sx**: ...
// - **sxx**: ...
// - **sxy**: ...
// - **sy**: ...
// - **errors**: ...
// - **x**: ...
// - **y**: ...
// - **y\_hat**: ...
//
// ## Parameters
// - tables: Input data. Default is piped-forward data (`<-`).
//
// ## Examples
//
// ### Perform a linear regression on a dataset
// ```
// import "contrib/anaisdg/statsmodels"
// import "sampledata"
//
// < sampledata.float()
// >     |> statsmodels.linearRegression()
// ```
~~~

@Anaisdg, any chance you could shed some light on what these columns represent?

Reply:
sx = sum of x 
sxx = sum of x squared
sxy = sum of x*y
sy = sum of y
y = field value
y_hat = linear regression values
errors = residual sum of squares defined by (r.y - r.y_hat) ^ 2.0 in this context

Reply:
@Anaisdg: how would you describe `x` here? (If `y` is field value).

Reply:
@ x is an index [1,2,3,4...n] with the assumption that the timestamps are regularly spaced.